File: app/src/main/java/org/digitalcampus/oppiamobile/utils/views/NoSwipeableViewpager.java
package org.digitalcampus.oppiamobile.utils.views;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.viewpager.widget.ViewPager;

public class NoSwipeableViewpager extends ViewPager {

    private boolean canScroll;

    public NoSwipeableViewpager(Context context) {
        super(context);
        canScroll = false;
    }
    public NoSwipeableViewpager(Context context, AttributeSet attrs) {
        super(context, attrs);
        canScroll = false;
    }
    public void setCanScroll(boolean canScroll) {
        this.canScroll = canScroll;
    }
    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return canScroll && super.onTouchEvent(ev);
    }
    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return canScroll && super.onInterceptTouchEvent(ev);
    }

}

File: app/src/main/java/org/digitalcampus/oppiamobile/utils/views/NoSwipeableViewpager.kt
package org.digitalcampus.oppiamobile.utils.views

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import androidx.viewpager.widget.ViewPager

class NoSwipeableViewpager : ViewPager {
    private var canScroll: Boolean

    constructor(context: Context?) : super(context!!) {
        canScroll = false
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    ) {
        canScroll = false
    }

    fun setCanScroll(canScroll: Boolean) {
        this.canScroll = canScroll
    }

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return canScroll && super.onTouchEvent(ev)
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return canScroll && super.onInterceptTouchEvent(ev)
    }
}